src/System.Net.Security/src/System/Net/Unix/SslConnectionInfo.cs

```diff
@@ -74,12 +80,19 @@ private static ExchangeAlgorithmType MapExchangeAlgorithmType(long mac)
             {
                 case 1:
                     return ExchangeAlgorithmType.RsaKeyX;
-                case 2 | 32:
-                    return ExchangeAlgorithmType.RsaSign;
-                case 128:
+                case 2:
+                case 4:
+                case 8:
                     return ExchangeAlgorithmType.DiffieHellman;
+                case 16:
+                case 32:
+                case 64:
+                case 128:
+                case 256:
+                case 512:
+                case 1024:
+                    return (ExchangeAlgorithmType) mac;
```

Comment:
> if OpenSSL exposes a concept Windows doesn't

@shrutigarg can confirm this. I believe even on Windows, values outside the defined enum set are possible. Your point about updating the enums still stands but I wanted to clarify that it is not specific to OpenSSL